miden-lib/asm/shared/util/account_id.masm

```diff
@@ -0,0 +1,387 @@
+const.MAX_NUM_PROCEDURES=256
+
+# Bit pattern for a faucet account, after the account type mask has been applied.
+const.FAUCET_ACCOUNT=0x20 # 0b10_0000
+
+# Bit pattern for an account w/ updatable code, after the account type mask has been applied.
+const.REGULAR_ACCOUNT_UPDATABLE_CODE=0x10 # 0b01_0000
+
+# Bit pattern for an account w/ immutable code, after the account type mask has been applied.
+const.REGULAR_ACCOUNT_IMMUTABLE_CODE=0 # 0b00_0000
+
+# Bit pattern for a fungible faucet w/ immutable code, after the account type mask has been applied.
+const.FUNGIBLE_FAUCET_ACCOUNT=0x20 # 0b10_0000
+
+# Bit pattern for a non-fungible faucet w/ immutable code, after the account type mask has been
+# applied.
+const.NON_FUNGIBLE_FAUCET_ACCOUNT=0x30 # 0b11_0000
+
+# Given the least significant 32 bits of an account id's first felt, this mask defines the bits used
+# to determine the account type.
+const.ACCOUNT_ID_TYPE_MASK_U32=0x30 # 0b11_0000
+
+# Given the least significant 32 bits of an account id's first felt, this mask defines the bits used
+# to determine the account version.
+const.ACCOUNT_VERSION_MASK_U32=0x0f # 0b1111
+
+# Given the least significant 32 bits of an account ID's first felt, this mask defines the bits used
+# to determine the account storage mode.
+const.ACCOUNT_ID_STORAGE_MODE_MASK_U32=0xC0 # 0b1100_0000
+
+# Given the least significant 32 bits of an account ID's first felt with the storage mode mask
+# applied, this value defines the public storage mode.
+const.ACCOUNT_ID_STORAGE_MODE_PUBLIC_U32=0 # 0b0000_0000
+
+# Given the least significant 32 bits of an account ID's first felt with the storage mode mask
+# applied, this value defines the private storage mode.
+const.ACCOUNT_ID_STORAGE_MODE_PRIVATE_U32=0x80 # 0b1000_0000
+
+# CONSTANT ACCESSORS
+# =================================================================================================
+
+#! Returns the account storage slot at which faucet data is stored.
+#! Fungible faucet: The faucet data consists of [0, 0, 0, total_issuance]
+#! Non-fungible faucet: The faucet data consists of SMT root containing minted non-fungible assets.
+#!
+#! Inputs:  []
+#! Outputs: [faucet_storage_data_slot]
+#!
+#! Where:
+#! - faucet_storage_data_slot is the account storage slot at which faucet data is stored.
+export.get_faucet_storage_data_slot
+    push.FAUCET_STORAGE_DATA_SLOT
+end
+
+#! Returns the maximum number of account storage slots.
+#!
+#! Inputs:  []
+#! Outputs: [max_num_storage_slots]
+#!
+#! Where:
+#! - max_num_storage_slots is the maximum number of account storage slots.
+export.get_max_num_storage_slots
+    push.MAX_NUM_STORAGE_SLOTS
+end
+
+#! Returns the maximum number of account interface procedures.
+#!
+#! Inputs:  []
+#! Outputs: [max_num_procedures]
+#!
+#! Where:
+#! - max_num_procedures is the maximum number of account interface procedures.
+export.get_max_num_procedures
+    push.MAX_NUM_PROCEDURES
+end
```

Comment:
These procedures are not really related to account ID and, from what I can tell, don't need to be shared. If so, I'd move them back to the kernel's `accounts.masm` file.